testing/test_newhooks.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+
+class TestHooks:
+
+    @pytest.fixture(autouse=True)
+    def create_test_file(self, testdir):
+        testdir.makepyfile("""
+            import os
+            def test_a(): pass
+            def test_b(): pass
+            def test_c(): pass
+        """)
+
+    def test_runtest_logreport(self, testdir):
+        """Test that log reports from pytest_runtest_logreport when running
+        with xdist contain a "node" attribute. (#8)
+        """
+        testdir.makeconftest("""
+            def pytest_runtest_logreport(report):
+                if hasattr(report, 'node'):
+                    slaveid = report.node.slaveinput['slaveid']
+                    if report.when == "call":
+                        print("HOOK: %s %s" % (report.nodeid, slaveid))
+        """)
+        res = testdir.runpytest('-n1', '-s')
```

Comment:
A comment: I tried using `inline_run` just to get the recorded hook calls. It worked fine when running just this test, but failed when executed together with the rest of the test suite: it would not record any hook calls. No idea why.